src/appleseed.python/bindblendermeshconverter.cpp

```diff
@@ -0,0 +1,222 @@
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+// appleseed.renderer headers.
+#include "renderer/api/object.h"
+
+// appleseed.foundation headers.
+#include "foundation/platform/python.h"
+
+namespace bpy = boost::python;
+using namespace renderer;
+
+// Blender data structures
+struct MeshFace
+{
+    unsigned int v[4];
+    short mat_nr;
+    char edcode, flag;
+};
+
+struct MeshVert
+{
+    float co[3];
+    short no[3];
+    char flag, bweight;
+};
+
+struct MeshTexFace
+{
+    float uv[4][2];
+    void *tpage;
+    char flag, transp;
+    short mode, tile, unwrap;
+};
+
+
+// The following function takes a series of pointers to Blender mesh data and modifies the appleseed MeshObject entity
+//      blender_object = the appleseed MeshObject created earlier in the export process
+//      vertices_length = the number of vertices in the mesh
+//      vertex_pointer = a string pointer to the first element of the vertex array
+//      Same template applies to mesh faces
+
+void convert_bl_mesh(
+    MeshObject*                 blender_mesh,
+    const size_t                vertices_length,
+    const size_t                vertex_pointer,
+    const size_t                triangle_length,
+    const size_t                triangle_pointer,
+    const size_t                uv_layer_pointer,
+    const bool                  export_normals,
+    const bool                  export_uvs)
+{
+    // Convert size_t numbers to actual pointers
+    MeshVert* vertices = reinterpret_cast<MeshVert *>(vertex_pointer);
+    MeshFace* triangles = reinterpret_cast<MeshFace *>(triangle_pointer);
+    MeshTexFace* uv_face = reinterpret_cast<MeshTexFace *>(uv_layer_pointer);
+
+    blender_mesh->reserve_vertices(vertices_length);
+    blender_mesh->reserve_triangles(triangle_length);
+
+    if (export_normals == true)
+    {
+        blender_mesh->reserve_vertex_normals(vertices_length);
+    }
+
+    if (export_uvs == true)
+    {
+        blender_mesh->reserve_tex_coords(vertices_length);
+    }
+
+    // Push vertices
+    for (size_t vertex_index = 0; vertex_index < vertices_length; ++vertex_index)
+    {
+        MeshVert &vert = vertices[vertex_index];
```

Comment:
Ampersands and stars go on the type's side: `MeshVert&`.

Fix everywhere.

Reply:
Make const: `const MeshVert& vert`.

In general, make `const` everything that can be.

Fix everywhere.